src/fmu/dataio/_metadata.py

```diff
@@ -99,8 +102,74 @@ def _get_meta_objectdata(
     )
 
 
-def _get_meta_access(access: dict) -> meta.SsdlAccess:
-    return meta.SsdlAccess.model_validate(access)
+def _get_meta_access(dataio) -> dict | None:
```

Comment:
Expanded this a lot, and put the assembling of the `access` block here instead of just passing the `config.access`. Hopefully, this is one step towards removing the pattern of updating the config and carrying it around during runtime.

This should possibly/probably be done with Pydantic logic instead of making everything, then passing it to Pydantic for validation. But for now this makes it a bit more feasible to make and read 🤷‍♂️